packages/protocol/docs/pacaya-to-shasta-transition.md

```diff
@@ -0,0 +1,62 @@
+  _The Taiko preconfer is likely the best entity to activate the inbox before submitting the first proposal. We can remove the rest of the preconfers during this period to ensure no one else proposes before the inbox is activated_  
+  _Forced inclusions submitted before the fork and not processed before it will be lost, and need to be submitted again to the new inbox_
+
+- **Anchor**: The Shasta [anchor contract](../contracts/layer2/core/Anchor.sol) will be deployed as an upgradable contract to a new address, but the entrypoint is kept using the [AnchorForkRouter](../contracts/layer2/core/AnchorForkRouter.sol). This fork router will be deployed to the existing Anchor address and route to the old and new implementation based on the function signature. Calls to the `anchorV3` will be routed to the Pacaya anchor and calls to `anchorV4` will be routed to the new shasta anchor.
+  Preconfers need to deposit their bonds into this contract before their first turn to propose after `FORK_TIMESTAMP` to avoid their proposal being treated as a low bond proposal.
+  **The contract does not handle fork logic based on the timestamp, and it is up to the preconfer to decide which function to call based on the timestamp.**
+  _Deposits and withdrawals will be available via the existing address_
+
+- **Signal Service**: The [SignalService](../contracts/shared/signal/SignalService.sol) is deployed as an upgradable contract to a new address, but a fork router will be used. The new SignalService is abi compatible for sending and receiving signals with the Pacaya implementation to ensure backwards compatibility for bridges.  
+  But between the time the new version is deployed and the fork activated we need to keep verification working with the old checkpoint mechanism. Because of that [SignalForkRouter](../contracts/shared/signal/SignalServiceForkRouter.sol) will be deployed to the existing SignalService address and redirect calls based on timestamp. Before `FORK_TIMESTAMP` it will send calls to the Pacaya implementation, and afterwards to the shasta implementation.
+  _Note that signals sent before the fork will still be available and can be verified, but a new merkle proof needs to be generated(this is because the checkpoints are different so the old proof won't pass validation)_
+
+## Steps
+
+1. **Deploy Shasta contracts**
+   Deploy Shasta inbox, anchor, signal service and other shasta contracts. Set `_genesisBlockHash = 0` on the inbox at deployment time. This sets the inbox to an empty state. **No proposals should happen to the inbox at this point**.
+   The `FORK_TIMESTAMP` should already be set to the shasta fork timestamp.  
+   **After this point proposers can deposit their bonds to the `BondManager` contract on L2**.
+
+2. **Deploy the Pacaya contracts**
+   Since we'll upgrade the contracts to use the fork router, we need to deploy the pacaya contracts(Inbox, Anchor and SignalService) to a new address so the router can forward requests to them.
```

Comment:
Probly we don't need to deploy the Pacaya contracts(Inbox, Anchor and SignalService) to new addresses,
- For `Inbox`, we only need to upgrade the old Pacaya inbox, if we decided to add the `timestamp` checks
- For `Anchor` and `SignalService`,  we need to upgrade the impl address of the current proxy to the new ForkRouter (with `oldFork == oldPacayaImplAddress`) impl address